reactor-netty-http/src/main/java/reactor/netty/http/client/Http2ConnectionLivenessHandler.java

```diff
@@ -0,0 +1,285 @@
+ * limitations under the License.
+ */
+package reactor.netty.http.client;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.handler.codec.http2.Http2ConnectionEncoder;
+import io.netty.handler.codec.http2.Http2PingFrame;
+import reactor.util.Logger;
+import reactor.util.Loggers;
+import reactor.util.annotation.Nullable;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+/**
+ * Handler that supports connection health checks using HTTP/2 Ping Frames.
+ *
+ * <p>This Handler sends a ping frame at the specified interval when no frame is being read or written,
+ * ensuring the connection health is monitored. If a ping ACK frame is not received within the configured interval,
+ * the connection will be closed.</p>
+ *
+ * <p>Ping frame checking will not be performed while a read or write operation is in progress.</p>
+ *
+ * <p>Be cautious when setting a very short interval, as it may cause the connection to be closed,
+ * even if the keep-alive setting is enabled.</p>
+ *
+ * <p>If no interval is specified, no ping frame checking will be performed.</p>
+ *
+ * @author raccoonback
+ * @since 1.2.3
+ */
+final class Http2ConnectionLivenessHandler extends ChannelDuplexHandler {
+
+	private static final Logger log = Loggers.getLogger(Http2ConnectionLivenessHandler.class);
+
+	private ScheduledFuture<?> pingScheduler;
+
+	private final ChannelFutureListener pingWriteListener = new PingWriteListener();
+	private final Http2ConnectionEncoder encoder;
+	private final long pingAckTimeoutNanos;
+	private final long pingScheduleIntervalNanos;
+	private final int pingAckDropThreshold;
+
+	private int pingAckDropCount;
+	private long lastSentPingData;
+	private long lastReceivedPingTime;
+	private long lastSendingPingTime;
+	private long lastIoTime;
+	private boolean isPingAckPending;
+
+	public Http2ConnectionLivenessHandler(Http2ConnectionEncoder encoder, @Nullable Duration pingAckTimeout,
+	                                      @Nullable Duration pintScheduleInterval, @Nullable Integer pingAckDropThreshold) {
+		Objects.requireNonNull(encoder, "encoder");
+
+		this.encoder = encoder;
+
+		if (pingAckTimeout != null) {
+			this.pingAckTimeoutNanos = pingAckTimeout.toNanos();
+		}
+		else {
+			this.pingAckTimeoutNanos = 0L;
+		}
+
+		if (pintScheduleInterval != null) {
+			this.pingScheduleIntervalNanos = pintScheduleInterval.toNanos();
+		}
+		else {
+			this.pingScheduleIntervalNanos = 0L;
+		}
+
+		if (pingAckDropThreshold != null) {
+			this.pingAckDropThreshold = pingAckDropThreshold;
+		}
+		else {
+			this.pingAckDropThreshold = 0;
+		}
```

Comment:
It initializes the `pingAckTimeoutNanos`, `pingScheduleIntervalNanos`, and `pingAckDropThreshold` values provided by the user.